### ISSUE TITLE
[DART] Add Dart client option to NOT convert model property names to camel case

### ISSUE THREAD

**Issue by jackvz** [swagger-api/swagger-codegen#9384]
##### Description

Model property names are converted to camel case. There should be an option to use the original case.

Converting the model property names to camel case does not match the naming convention in the OpenAPI spec that I am using, [here](https://raw.githubusercontent.com/jackvz/mezzanine-cartridge-api/master/swagger.json). This breaks, and so for any property names that are not served with camel case names, in their original format, no data is matched, and an error results.

##### Swagger-codegen version

v3.0.7 I believe. I pulled this from the master branch today (18 Apr)

##### Swagger declaration file content or url

[The definition that I am working with](https://raw.githubusercontent.com/jackvz/mezzanine-cartridge-api/master/swagger.json). Note: This is not a OpenAPI / Swagger definition issue. I think we need to add an option to NOT convert model property names to camel case.

##### Command line used for generation

This is based on [bin/dart-petstore.sh](https://github.com/swagger-api/swagger-codegen/blob/master/bin/dart-petstore.sh). I am only generating a non-browser client, so everything after that is cleared.

java $JAVA_OPTS -jar $executable generate -t modules/swagger-codegen/src/main/resources/dart -i https://raw.githubusercontent.com/jackvz/mezzanine-cartridge-api/master/swagger.json -l dart -o client/mezzanine_dart_client -DhideGenerationTimestamp=true -DbrowserClient=false -DcamelCaseModelProperties=false -DpubName=mezzanine_dart_client -DpubDescription="A Dart client for the REST Web API for Mezzanine with Cartridge"

##### Steps to reproduce

Clone this repo.

Replace line 35 in [bin/dart-petstore.sh](https://github.com/swagger-api/swagger-codegen/blob/master/bin/dart-petstore.sh) with:

java $JAVA_OPTS -jar $executable generate -t modules/swagger-codegen/src/main/resources/dart -i https://raw.githubusercontent.com/jackvz/mezzanine-cartridge-api/master/swagger.json -l dart -o client/mezzanine_dart_client -DhideGenerationTimestamp=true -DbrowserClient=false -DcamelCaseModelProperties=false -DpubName=mezzanine_dart_client -DpubDescription="A Dart client for the REST Web API for Mezzanine with Cartridge"

Note how all of the generated Dart models, for example client/mezzanine_dart_client/lib/model/category, have property names that are camel cased, and here is the real issue: It also tries to find the data in camel case, in the models' fromJsons methods, and fails, errorring out.

##### Related issues/PRs

None that I could find specifically for Dart. I am putting in a PR that will resolve this.

##### Suggest a fix/enhancement

I suggest adding a Dart client option to NOT convert model property names to camel case, and am putting in a PR

**Comment by nickmeinhold** [comment]
Thanks for your work on trying to solve this @jackvz 

Am I right in assuming your PR will create identifiers in `lowercase_with_underscores` format? 

I imagine you're aware of the Effective Dart guideline [DO name other identifiers using lowerCamelCase](
https://dart.dev/guides/language/effective-dart/style#do-name-other-identifiers-using-lowercamelcase)?  But anyway, that's better than having client code that doesn't work or going through and manually changing the fromJson keys back to `lowercase_with_underscores`, which is where I'm stuck at.  

I noticed #4898 discusses how parameter names get mangled to conform to the conventions of the language and I believe #5019 was supposed to be a fix for Dart, but the problem looks to still not be solved.  

I've tried digging around in the mustache files to see if I can figure out how to get the unmangled `baseName` to be used as the json keys in the models but no luck yet.  

I may be out of my league as I don't have any previous experience with this stuff but I thought I would check in and see if it seemed like a reasonable approach and if you thought it was worth pursuing.
